Completed Solution and Test File for #16 Credit Card Number Challenge

Ammmoe opened the pull request:
Hi Team Members!

I wrote the `check_credit` python module and the `test_check_credit` unittest module for `Checking Credit Card Number Challenge`.

Please review and suggest me for any improvement points!

If you are OK with the files, then this resolves the challenge #16 .

Ammmoe committed:
completed check_credit solution and test python files

Ammmoe committed:
Fixes formatting issue with Ruff

Ammmoe committed:
Fixes the creation date from 2024 to 2025

KarimMakki commented on solutions/check_credit.py:
```diff
@@ -0,0 +1,90 @@
+
+
+# ---- define function ----
+def check_credit(card_number: str) -> str:
+    """Checks if a credit card number is a valid American Express, MasterCard, or Visa card number
+
+    Args:
+        card_number (str): the credit card number to check
+
+    Returns:
+        str: return 'AMEX', 'MASTERCARD', 'VISA', or 'INVALID' according to the validity and the structure of card number
+
+    Raises: AssertionError
+        - if card_number is not a string
+
+    Raises: ValueError
+        - if card_number contains other than numeric characters (0-9)
+
+    >>> check_credit('378282246310005')
+    'AMEX'
+
+    >>> check_credit('5555555555554444')
+    'MASTERCARD'
+
+    >>> check_credit('4111111111111111')
+    'VISA'
+
+    >>> check_credit('1234567890')
+    'INVALID'
+    """
```
**The function name and docstring need to be refactored as follows:**  
1. The function name could be `check_credit_card`, as it would be a more descriptive name, don't forget to change the test file name to match the function file name.
2. I believe you could add an important assumption for the argument to provide better context:  
   `The credit card number to validate. Must be a string containing only digits (0-9), with an appropriate length for the card type.`  
3. The `Raises` section can be formatted better. You could structure it as follows:  
   ```python
   Raises:
       AssertionError: If `card_number` is not a string.
       ValueError: If `card_number` contains other than numeric digits (0-9).

Ammmoe replied:
I agree with your suggestions and have modified my files accordingly!

KarimMakki commented on solutions/check_credit.py:
```diff
@@ -0,0 +1,90 @@
+    >>> check_credit('1234567890')
+    'INVALID'
+    """
+    # Defensive error catching
+    assert isinstance(card_number, str), "Card number must be formatted as a string"
+
+    if not card_number.isdigit():
+        raise ValueError("Card number must only contain numeric characters (0-9)")
```
The comment for the error catching could be like this to explain more about the strategy (the "why"):
`# Ensure the input is in the correct format and contains only numbers`

Ammmoe replied:
Thanks! I've modified accordingly.

KarimMakki reviewed (changes requested):
Your solution looks good overall! I’ve added some remarks as conversations, which include:

1. Changing function and test file names for better clarity.
2. Improving function docstrings.
3. Modfiying comments to better explain the strategy (the "why").

Ammmoe committed:
refactor: changed function and test file names, improved function docstring, and modified comments

Ammmoe commented:
Thanks for your review and insights @KarimMakki !

I've refactored the followings according to your suggestions:

1. Added the assumption regarding the argument in the function docstring
2. Reformatted the raises session in the function docstring
3. Changed the names of the function and unittest modules to `check_credit_card` and `test_check_credit_card` respectively 
4. Modified the comment for the defensive error catching mechanism

Please have a look and let me know if there are more changes that I should make.

Ammmoe committed:
Removed strategy from function documentation

KarimMakki reviewed (approved):
Great job, @Ammmoe ! The modifications you made are spot on, and everything looks good now.
